driver/fvGFS/fv_ufs_restart_io.F90

```diff
@@ -132,6 +132,7 @@ subroutine fv_dyn_restart_register (Atm)
    ! srf_wnd
    nvar2d_srf_wnd = 2
    allocate (srf_wnd_var2(nx,ny,nvar2d_srf_wnd), srf_wnd_var2_names(nvar2d_srf_wnd))
```

Comment:
Do any of the UFS applications use fv_srf_wind.res? This is used for AM4 (I believe) but not in SHiELD.

Reply:
All UFS applications use it, if they enable quilting restart. From atmos_model.F90 subroutine atmos_model_init:

```f90
   if (quilting_restart) then
      call fv_dyn_restart_register (Atm(mygrid))
      call fv3atm_restart_register (GFS_data%Sfcprop, GFS_restart_var, Atm_block, GFS_control)
   endif
```

Reply:
I will ping @DusanJovic-NOAA for comment on this.

Reply:
The fv_srf_wnd.res restart file is always written:

https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/dev/emc/tools/fv_io.F90#L847-L858

regardless on which UFS configuration is used.

However in read routine:

https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/dev/emc/tools/fv_io.F90#L486-L498

I see that if the file is not present in the run directory it will print warning and continue execution. Whether this file is needed to get b4b identical restart run compared to non-restart I'm not sure, I guess we can run test to see what will happen if we do not create this file. If the restart is b4b,  maybe we can skip writing/reading this file in UFS.

Reply:
Regardless of whether the file is written, you should initialized your arrays.

Reply:
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/dev/emc/tools/fv_io.F90#L692-L693

Is `else` missing between line 692 and 693?

Reply:
I should mention that this is relevant to this PR I just noticed this while looking at where and how fv_srf_wnd.res is created and used.

Reply:
@DusanJovic-NOAA - It looks like it might be missing.  Please open an issue detailing your concern with this section of tools/fv_io.F90